src/lib/list/selection-list.ts

```diff
@@ -0,0 +1,228 @@
+  template: '<ng-content></ng-content>',
+  styleUrls: ['list.css'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSelectionList implements AfterContentInit, OnDestroy {
+  private _disableRipple: boolean = false;
+
+  private _disabled: boolean = false;
+
+  /** Tab index for the selection-list. */
+  _tabIndex = 0;
+
+  /** Track which options we're listening to for focus/destruction. */
+  private _subscribed: WeakMap<MdListOption, boolean> = new WeakMap();
+
+  /** Subscription to tabbing out from the selection-list. */
+  private _tabOutSubscription: Subscription;
+
+  /** Subscription to option changes from the selection-list. */
+  private _optionSubscription: Subscription;
+
+  /** Whether or not the option is selectable. */
+  protected _selectable: boolean = true;
+
+  /** The FocusKeyManager which handles focus. */
+  _keyManager: FocusKeyManager;
+
+  /** The option components contained within this selection-list. */
+  @ContentChildren(MdListOption) options;
+
+  /** options which are selected. */
+  selectedOptions: SelectionModel<MdListOption> = new SelectionModel<MdListOption>(true);
+
+  /**
+   * Whether the ripple effect should be disabled on the list-items or not.
+   * This flag only has an effect for `mat-selection-list` components.
+   */
+  @Input()
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
+
+  /** Whether the selection-list is disabled */
+  @Input()
+  get disabled() { return this._disabled; }
+  set disabled(value: any) { this._disabled = coerceBooleanProperty(value); }
+
+  constructor(private _element: ElementRef) { }
+
+  ngAfterContentInit(): void {
+    this._keyManager = new FocusKeyManager(this.options).withWrap();
+
+    // Prevents the selection-list from capturing focus and redirecting
+    // it back to the first option when the user tabs out.
+    this._tabOutSubscription = this._keyManager.tabOut.subscribe(() => {
+      this._tabIndex = -1;
+      setTimeout(() => this._tabIndex = 0);
+    });
+
+    // Go ahead and subscribe all of the initial options
+    this._subscribeOptions(this.options);
+    // When the list changes, re-subscribe
+    this._optionSubscription =
+      this.options.changes.subscribe((options: QueryList<MdListOption>) => {
+        this._subscribeOptions(options);
+      });
+  }
+
+  ngOnDestroy(): void {
+    if (this._tabOutSubscription) {
+      this._tabOutSubscription.unsubscribe();
+    }
+
+    if (this._optionSubscription) {
+      this._optionSubscription.unsubscribe();
+    }
+  }
+
+  /**
+   * Whether or not this option is selectable. When a option is not selectable,
+   * it's selected state is always ignored.
+   */
+  @Input()
+  get selectable(): boolean { return this._selectable; }
+  set selectable(value: boolean) {
+    this._selectable = coerceBooleanProperty(value);
+  }
```

Comment:
I think this `selectable` property is in the wrong class, but also not necessary for `option` and should probably be removed.